lib/iris/common/metadata.py

```diff
@@ -289,7 +323,16 @@ def MetadataManagerFactory(cls, **kwargs):
         fields will default to a value of 'None'.
 
     """
+    # Restrict factory to appropriate metadata classes only.
```

Comment:
Why is this repeating the same section already in `__init__`?

Reply:
Good spot.
I removed the one in the __init__ : it is only ever called from the factory function itself, so let's just check once there.